addons/account/tests/test_account_move_taxes_edition.py

```diff
@@ -0,0 +1,217 @@
+        Test the amount of tax account.move.line is cumulative for the same tax.
+        This test uses the following scenario:
+            - Create manually a debit line of 1000 having a tax.
+            - Assume a line containing the tax amount is created automatically.
+            - Edit the tax line amount of the auto-generated line by adding 5.
+            - Create manually a credit line to balance the two previous lines.
+            - Save the move.
+            - Edit the move.
+            - Create manually a debit line of 2000 having the same tax.
+            - Assume the line containing the tax amount has been updated (no new line created).
+            - Create manually a credit line to balance the four previous lines.
+            - Save the move.
+
+            DEBIT       |   CREDIT
+            ----------------------
+            1000        |
+            2000        |
+            tax(3000)   |
+                        |   1005 + tax(1000)
+                        |   1995 + tax(2000)
+        '''
+        account = self.env['account.account'].search([('deprecated', '=', False)], limit=1)
+        tax = self.env['account.tax'].search([], limit=1)
+        journal = self.env['account.journal'].search([], limit=1)
+
+        with Form(self.env['account.move'], view=self.env.ref('account.view_move_form')) as move_form:
+            move_form.ref = 'azerty'
+            move_form.journal_id = journal
+
+            # Create a new account.move.line with debit amount.
+            with move_form.line_ids.new() as debit_line:
+                debit_line.account_id = account
+                debit_line.debit = 1000
+                debit_line.tax_ids.clear()
+                debit_line.tax_ids.add(tax)
+
+                self.assertTrue(debit_line.recompute_tax_line)
+
+            # Edit the tax account.move.line
+            with move_form.line_ids.edit(index=1) as tax_line:
+                tax_line.debit += 5
+
+            amount = tax.compute_all(1000)['total_included']
+
+            # Create a third account.move.line with credit amount.
+            with move_form.line_ids.new() as credit_line:
+                credit_line.account_id = account
+                credit_line.credit = amount + 5
+
+            move = move_form.save()
+
+            self.assertEqual(len(move.line_ids), 3)
+
+        with Form(move, view=self.env.ref('account.view_move_form')) as move_form:
+            # Create a new account.move.line with debit amount.
+            with move_form.line_ids.new() as debit_line2:
+                debit_line2.account_id = account
+                debit_line2.debit = 2000
+                debit_line2.tax_ids.clear()
+                debit_line2.tax_ids.add(tax)
+
+                self.assertTrue(debit_line2.recompute_tax_line)
+
+            amount = tax.compute_all(2000)['total_included']
+
+            with move_form.line_ids.new() as credit_line2:
```

Comment:
that last part of the 3th test is unclear.

So far, you should have


Name | debit | credit | tax_ids | tax_line_id
------------ | ------------- | ------------- | ------------- | -------------
debit line 1 | 1000 | | 10% |
tax line | 105 |  | | 10% 
credit line 1 |  | 1105 | |

When adding the second debit line, assert the debit amount of tax line is set to 300 and assert the number of lines is now 4 (no fifth line added)

Then set the last credit line 2 manually to (3300 - 1105) and check the whole move is now

Name | debit | credit | tax_ids | tax_line_id
------------ | ------------- | ------------- | ------------- | -------------
debit line 1 | 1000 | | 10% |
tax line | 300 |  | | 10% 
credit line 1 |  | 1105 | |
debit line 2 | 2000 | | 10% | 
credit line 2 |  | 2195 | |